polkadot/xcm/pallet-xcm/src/lib.rs

```diff
@@ -1193,14 +1197,89 @@ impl<T: Config> QueryHandler for Pallet<T> {
 	}
 }
 
+#[derive(Copy, Clone, PartialEq, Debug)]
+enum TransferType {
```

Comment:
This could also be exposed by `xcm-executor` instead of `pallet-xcm`. WDYT?

Basically expose it directly through newly added `AssetTransferFilter` trait bound added to `XcmExecutor`.

Reply:
that will work

Reply:
done https://github.com/paritytech/polkadot-sdk/pull/1672/commits/931e09371562d6ffaa2cb7fb09b9a43ea5901e0a